subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/adapter/ProtocolToModelAdapter.java

```diff
@@ -758,14 +770,15 @@ public void invoke(MethodInvocation invocation) throws Throwable {
         ViewDecoration restrictTo(Set<Class<?>> viewTypes);
     }
 
-    private static class NoOpDecoration implements ViewDecoration, Serializable {
+    @VisibleForTesting
+    static class NoOpDecoration implements ViewDecoration, Serializable {
         @Override
         public void collectInvokers(Object sourceObject, Class<?> viewType, List<MethodInvoker> invokers) {
         }
 
         @Override
         public boolean equals(Object obj) {
-            return obj instanceof NoOpDecoration;
+            return this == obj || obj != null && obj.getClass() == getClass();
```

Comment:
:o: Shouldn't the `obj != null && obj.getClass() == getClass()` be wrapped in parenthesis so it is evaluated as one statement?